server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -47,172 +45,208 @@
     }
 
     @Override
     public SetMessagesResponse process(SetMessagesRequest request, MailboxSession mailboxSession) {
-        Mailbox outbox;
+        Builder responseBuilder = SetMessagesResponse.builder();
+        request.getCreate()
+            .stream()
+            .forEach(create -> handleCreate(create, responseBuilder, mailboxSession));
+        return responseBuilder.build();
+    }
+
+    private void handleCreate(CreationMessageEntry create, Builder responseBuilder, MailboxSession mailboxSession) {
         try {
-            outbox = getOutbox(mailboxSession).orElseThrow(() -> new MailboxRoleNotFoundException(Role.OUTBOX));
-        } catch (MailboxException | MailboxRoleNotFoundException e) {
-            LOGGER.error("Unable to find a mailbox with role 'outbox'!");
-            throw Throwables.propagate(e);
-        }
+            validateImplementedFeature(create, mailboxSession);
+            validateArguments(create);
+            validateRights(create, mailboxSession);
+            MessageWithId created = handleOutboxMessages(create, mailboxSession);
+            responseBuilder.created(created.getCreationId(), created.getValue());
 
-        List<String> allowedSenders = ImmutableList.of(mailboxSession.getUser().getUserName());
+        } catch (MailboxUnhandledRequestException e) {
+            responseBuilder.notCreated(create.getCreationId(), 
+                    SetError.builder()
+                        .type("invalidProperties")
+                        .properties(MessageProperty.mailboxIds)
+                        .description("This mailbox can't handle this request")
+                        .properties()
+                        .build());
 
-        // handle errors
-        Predicate<CreationMessage> validMessagesTester = creationMessage -> creationMessage.isValid() && isAllowedFromAddress(creationMessage, allowedSenders);
-        Predicate<CreationMessage> invalidMessagesTester = validMessagesTester.negate();
-        Function<CreationMessage, List<ValidationResult>> toValidationResults = creationMessage -> ImmutableList.<ValidationResult>builder()
-            .addAll(creationMessage.validate())
-            .addAll(validationResultForIncorrectAddress(creationMessage, allowedSenders))
-            .build();
+        } catch (MailboxSendingNotAllowedException e) {
+            responseBuilder.notCreated(create.getCreationId(), 
+                    SetError.builder()
+                        .type("invalidProperties")
+                        .properties(MessageProperty.from)
+                        .description("Invalid 'from' field. Must be one of " + e.getAllowedFroms())
+                        .build());
 
-        SetMessagesResponse.Builder responseBuilder = SetMessagesResponse.builder()
-                .notCreated(handleCreationErrors(invalidMessagesTester, toValidationResults, request));
+        } catch (MailboxNotImplementedException e) {
```

Comment:
MailboxUnhandledRequestException was more a "it should not happen" exception, just dropped